### DISCUSSION TITLE
Welcome to AnduinOS Discussions!

### DISCUSSION THREAD

**Discussion by Anduin2017** [Anduin2017/AnduinOS#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below!

**Comment by quadransmuralis** [comment]
Hi there,
Just discovered AnduinOS - there's definitely room for a Jammy based distro using Gnome. (Trying to use Ubuntu has become such a struggle.)
Given your stated goal to welcome people from the world of Windows, you (in my humble opinion) must have an app store to make life easier for these people. I know Gnome Software app store is easy to download and install - IF you already have experience of Linux.
Good luck!

**Reply by Anduin2017** [reply]
Yeah we are working on this. Maybe based on flatpak.

**Reply by quadransmuralis** [reply]
When deactivating the Arc menu extension, the Power off letters in the icon in Gnome shell are too small.

Bravo! I really like AnduinOS. It looks modern. It seems reliable. I don't have to struggle with it as I do with Ubuntu or Debian.

**Reply by quadransmuralis** [reply]
Sorry - it's not the letters, it's the button in the middle of the Power off icon.
The same is true of the Suspend button.

**Reply by Anduin2017** [reply]
![image](https://github.com/user-attachments/assets/8e09ee8b-40b1-48e5-8ad2-6a8f8dc9b07a)

Is it?

**Reply by Anduin2017** [reply]
FYI. AnduinOS will be based on Noble very soon. (The Jammy verison will keep getting updates from Canonical till 2027)

Ref: https://github.com/Anduin2017/AnduinOS/discussions/34

**Comment by quadransmuralis** [comment]
[image: Screenshot from 2024-09-15 15-15-30.png]


On Sun, 15 Sept 2024 at 11:25, Anduin Xue ***@***.***> wrote:

> image.png (view on web)
> <https://github.com/user-attachments/assets/8e09ee8b-40b1-48e5-8ad2-6a8f8dc9b07a>
>
> Is it?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Anduin2017/AnduinOS/discussions/1#discussioncomment-10651002>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFP6GXXBIJOIT3UI4F5WD43ZWVHB7AVCNFSM6AAAAABLCVOVKWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANRVGEYDAMQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Robert Heath
+33 6 50 75 73 95

**Comment by Bosiux** [comment]
Hi @Anduin2017 , i've been using your OS for a couple of days and i met some issues tell me what you think:

- Executables on desktop cant be runned on double click
> After compiling my own C/C++/rust code on my Desktop  and assign the correct permissions, i cant double click to run it, only on one click, once side menu opens i can use "run as a program"
- System sometimes crashes
> After some certain events like running a buggy script (istantly) or closing and reopening your laptop several times, after logging  back in , the os block every IO and freeze the screen( forever or more than 4hrs, i havent tested more), once reboot you might lose up to 2 minutes from crash of file savings, downloads, executables (if your wondering how, ur not the only one).

**Reply by Anduin2017** [reply]
As for:Executables on desktop cant be runned on double click

I tested it with code:

```c
#include <gtk/gtk.h>

static void on_button_clicked(GtkWidget *widget, gpointer data) {
    g_print("Hello, World!\n");
}

void on_activate(GtkApplication *app, gpointer user_data);

int main(int argc, char *argv[]) {
    GtkApplication *app;
    int status;

    app = gtk_application_new("org.example.GtkApp", G_APPLICATION_FLAGS_NONE);

    g_signal_connect(app, "activate", G_CALLBACK(on_activate), NULL);

    status = g_application_run(G_APPLICATION(app), argc, argv);
    g_object_unref(app);

    return status;
}

void on_activate(GtkApplication *app, gpointer user_data) {
    GtkWidget *window;
    GtkWidget *button;

    window = gtk_application_window_new(app);
    gtk_window_set_title(GTK_WINDOW(window), "GTK 4 Example");
    gtk_window_set_default_size(GTK_WINDOW(window), 200, 100);

    button = gtk_button_new_with_label("Hello World");
    g_signal_connect(button, "clicked", G_CALLBACK(on_button_clicked), NULL);

    gtk_window_set_child(GTK_WINDOW(window), button);

    gtk_window_present(GTK_WINDOW(window));
}

```

And I built it with:

```bash
sudo apt-get install libgtk-4-dev
gcc `pkg-config --cflags gtk4` -o a.out a.c `pkg-config --libs gtk4`
chmod +x ./a.out
```

I found I can successfully lunch it via right click -> `Run as a program`. Indeed, double click is not lunching it.

I googled it and found:

In the Gnome desktop environment,the **ability to run any program by double-clicking on** it in the file manager is **deprecated**. It is indeed handy in some use cases. In other use cases, it can be confusing for users and for sure, it can be a security nightmare. Which is probably an important reason why it is deprecated.

It actually remains possible to set that up, however, it is not anymore set up by default.

 - If you are a developer needing to run an application during development, open a terminal. Enter the name of the program once (use <kbd>Tab</kbd> for command completion. Then use arrow up to repeat the command.
 - If you are an end user needing to run some application, then install it properly. Install or link the executable in your `~/bin` or `~/.local/bin` directory. Run it from the terminal just by typing the name, or create a launcher to it in `~/.local/share/applications`. Then, you can run it from the application overview, and even pin it to the dock for quick access.

Still, you could set up the ability to run files from the file manager by setting up a `.desktop` launcher that runs a file you provided as an argument. A `.desktop` launcher with the line `Exec=gnome-terminal -- %f` would open Gnome Terminal and (try to) run the file name provided, represented by the `%f`. If it is a script, you will see it running in the terminal. If it is a graphical application, Gnome terminal will briefly load, but disappear once the graphical application has started. A minimal `.desktop` launcher for this is:

    [Desktop Entry]
    Name=Generic application runner
    Comment=Runs executable files
    Exec=gnome-terminal -- %f
    Type=Application    

With the `.desktop` file installed in `~/.local/share/applications`, you can run any program by right-clicking and selecting "Open with...". You can associate files with a specific extension, e.g. `.sh` for scripts, to that launcher so it runs upon double click.

Alternatively, you will need to at least switch to a file manager still supporting the feature, or switch to another desktop environmemt like Mate or Xfce.

**Reply by Anduin2017** [reply]
Yes, actually, I occasionally encounter this issue myself while using AnduinOS. For example, I might mistakenly kill a critical process, a process might attempt to access a non-existent API, my memory might run out, or my virtual machine is starting up and it interferes with hardware that is currently suspended...

In many of these scenarios, the cause of the blockage often isn't the kernel itself but rather the specific hardware it relies on. It may be waiting on a process that cannot terminate. I believe if we need to address such issues, it could involve enhancing hardware timeout handling mechanisms, improving resource management to prevent system hangs, or building fail-safe protocols to gracefully handle critical process termination without freezing. Debugging and isolating faulty hardware behavior or adding kernel-level safeguards to detect and resolve hanging I/O operations more robustly might also help mitigate these situations.

However, in the short term, you can provide more information to help diagnose this issue. For example, you can perform kernel and system log analysis: after a reboot, use commands to check `/var/log/syslog`, `/var/log/kern.log`, or `dmesg` to see if there are any anomalies in the log records just before the system freeze. Pay particular attention to errors or warnings related to I/O operations, memory allocation, or file system access.

Memory leak or crash investigation is another area to consider: running unstable scripts might lead to memory or system resource leaks, causing resource exhaustion and ultimately freezing the system. Before running the problematic script, you can use tools like `htop`, `vmstat`, and `iostat` to monitor resource usage and check if there is memory exhaustion or heavy I/O load.

Additionally, you might consider checking for potential race conditions or deadlocks if the issue involves concurrent processes. Disabling or isolating potentially problematic hardware drivers temporarily, or using a controlled test environment to reproduce and capture system states, can also help narrow down the cause. In cases where frequent suspension and resumption are factors, you could test alternative power management configurations or kernel parameters to identify potential incompatibilities.

**Reply by Bosiux** [reply]
May I ask if this is an Ubuntu modded os or an entire os built with some Ubuntu modules? In case we would like to change something, do some kernel variations and compile it with our needings.from the repo code it seems a modded but idk.

**Reply by Anduin2017** [reply]
No it's not a Ubuntu modded OS. It's built from Debian's tool `debootstrap`. I chroot to a clean folder, downloaded Debian core packages, (including apt), then switch the source to Ubuntu, and downloaded the Ubuntu's kernel and apps.

**Comment by quadransmuralis** [comment]
Hello,

I've installed AnduinOS 1.3.2 on an 13th Gen Intel® Core™ i5-1334U × 12 laptop.

Using "Online accounts" apparently I have concluded the exchanges that should enable me to connect my google calendar - but the latter simply does not populate in the panel.

Can anyone kindly suggest something I should try?

Thanks.

**Reply by Anduin2017** [reply]
AnduinOS doesn't have a calendar app preinstalled. Please try `sudo apt install gnome-calendar`?

**Comment by Windows-10X** [comment]
Uhh. Does this have an arm64 version?

**Reply by Anduin2017** [reply]
Not yet. We are working on this. Hopefully 2026 04.

**Reply by Windows-10X** [reply]
Okay. Thanks for letting me know 😊

**Reply by Anduin2017** [reply]
It has NOW!

**Reply by Windows-10X** [reply]
YAYYYYYYY

**Comment by ghost** [comment]
Absolutely a great OS with its light, flexible and simple overall design. Ubuntu without the bloat and Im running on older hardware and it flies!

**Comment by RainGater** [comment]
Installed it today and looks polished and promising.  Great work!

Btw, I tried to change the RDP port from 3389 to a different one but couldn't make it happen.  Any ideas how to change the RDP port?  The following didn't work.

```
gsettings set org.gnome.desktop.remote-desktop.rdp port 55455

sudo ufw allow 55455/tcp
sudo ufw deny 3389/tcp

sudo ufw reload
```
